presto-main/src/main/java/com/facebook/presto/operator/TableFinishOperator.java

```diff
@@ -342,4 +351,200 @@ public void close()
     {
         Optional<ConnectorOutputMetadata> finishTable(Collection<Slice> fragments, Collection<ComputedStatistics> computedStatistics);
     }
+
+    public interface PartitionCommitter
+    {
+        void commitPartition(int partitionId, Collection<Slice> fragments);
+    }
+
+    private static class LifespanStageStateTracker
+    {
+        private final Map<LifespanStage, LifespanStageState> unrecoverableLifespanStageStates = new HashMap<>();
+        private final Map<LifespanStage, Map<Integer, LifespanStageState>> unfinishedRecoverableLifespanStageStates = new HashMap<>();
+        private final Map<LifespanStage, LifespanStageState> finishedRecoverableLifespanStages = new HashMap<>();
+
+        private final PartitionCommitter partitionCommitter;
+        private final JsonCodec<TableCommitContext> tableCommitContextCodec;
+
+        public LifespanStageStateTracker(PartitionCommitter partitionCommitter, JsonCodec<TableCommitContext> tableCommitContextCodec)
```

Comment:
nit: for private static inner class, you don't need `public` here for constructor.